v2/tools/generator/internal/config/configuration_merge_usage_example_test.go

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package config
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+// TestConfiguration_Merge_UsageExample demonstrates how to use the merge functionality
+func TestConfiguration_Merge_UsageExample(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	// Create a base configuration
+	base := NewConfiguration()
+	base.SchemaRoot = "schemas/"
+	base.Pipeline = GenerationPipelineAzure
+	base.AnyTypePackages = []string{"base-package1", "base-package2"}
+	base.EmitDocFiles = false
+
+	// Create another configuration to merge
+	other := NewConfiguration()
+	other.TypesOutputPath = "api/types" // This was empty in base, so it will be set
+	other.AnyTypePackages = []string{"other-package1", "other-package2"} // These will be appended
+	other.SamplesPath = "samples"
+
+	// Create configuration that demonstrates nested merging
+	other.SupportedResourcesReport = NewSupportedResourcesReport(other)
+	other.SupportedResourcesReport.OutputFolder = "reports"
+	other.SupportedResourcesReport.CurrentRelease = "v2.1.0"
+
+	// Perform the merge
+	err := base.Merge(other)
+	g.Expect(err).To(Succeed())
+
+	// Verify the results
+	g.Expect(base.SchemaRoot).To(Equal("schemas/")) // Preserved from base
+	g.Expect(base.Pipeline).To(Equal(GenerationPipelineAzure)) // Preserved from base
+	g.Expect(base.TypesOutputPath).To(Equal("api/types")) // Set from other
+	g.Expect(base.SamplesPath).To(Equal("samples")) // Set from other
+	g.Expect(base.EmitDocFiles).To(BeFalse()) // Preserved from base (false)
+	
+	// Verify slice was appended
+	g.Expect(base.AnyTypePackages).To(Equal([]string{
+		"base-package1", "base-package2", 
+		"other-package1", "other-package2",
+	}))
+
+	// Verify nested configuration was merged
+	g.Expect(base.SupportedResourcesReport).ToNot(BeNil())
+	g.Expect(base.SupportedResourcesReport.OutputFolder).To(Equal("reports"))
+	g.Expect(base.SupportedResourcesReport.CurrentRelease).To(Equal("v2.1.0"))
+
+	fmt.Printf("Merge successful! Final configuration:\n")
+	fmt.Printf("  SchemaRoot: %s\n", base.SchemaRoot)
+	fmt.Printf("  TypesOutputPath: %s\n", base.TypesOutputPath)
+	fmt.Printf("  SamplesPath: %s\n", base.SamplesPath)
+	fmt.Printf("  AnyTypePackages: %v\n", base.AnyTypePackages)
+	fmt.Printf("  ReportsOutputFolder: %s\n", base.SupportedResourcesReport.OutputFolder)
```

Comment:
Keep these lines as documentation, but comment them out - we don't want tests producing unnecessary output.